stdlib/public/runtime/BuiltinProtocolConformances.cpp

```diff
@@ -30,7 +30,39 @@ using StaticInfixWitness = SWIFT_CC(swift) bool(OpaqueValue *, OpaqueValue *,
 
 // Elf indirect symbol references.
 #if defined(__ELF__)
-#define INDIRECT_RELREF_GOTPCREL(SYMBOL) SYMBOL "@GOTPCREL + 1"
+#define INDIRECT_RELREF_GOTPCREL(SYMBOL) ".Lgot." SYMBOL " - . + 1"
+
+// Helper for defining the .type on linux.
+#if defined(__arm__) && !defined(__aarch64__)
+#define TYPE_OBJECT "%object"
+#else
+#define TYPE_OBJECT "@object"
+#endif
+
+// Helper to get the pointer sized pointer to symbol and the right alignment.
+#if defined(__x86_64__) || defined(__aarch64__)
+#define POINTER_FOR_GOT(SYMBOL) \
+  "  .quad (" SYMBOL ")\n" \
+  "  .size .Lgot." SYMBOL ", 8\n"
+
+#define P2ALIGN_FOR_GOT "  .p2align 3\n"
+#else
+#define POINTER_FOR_GOT(SYMBOL) \
+  "  .long (" SYMBOl ")\n" \
+  "  .size .Lgot." SYMBOL ", 4\n"
+
+#define P2ALIGN_FOR_GOT "  .p2align 2\n"
+#endif
+
+// Helper to significantly reduce the amount of code necessary to create got
+// equivalent variables for linux.
+#define GOT_EQUIVALENT(SYMBOL) \
+  __asm( \
+    "  .type .Lgot." SYMBOL ", " TYPE_OBJECT "\n" \
```

Comment:
I think that you should create another macro to generate the name rather than use the `.Lgot." SYMBOL"` in multiple locations.